sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/windowExpressions.scala

```diff
@@ -1014,3 +1014,77 @@ case class PercentRank(children: Seq[Expression]) extends RankLike with SizeBase
   override protected def withNewChildrenInternal(newChildren: IndexedSeq[Expression]): PercentRank =
     copy(children = newChildren)
 }
+
+/**
+ * Keep the Last non-Null value if any.
+ * MUST only be used in the Pandas API on Pyspark.
+ * input seq: null, 1, 2, 3, null, 4, 5, null
+ * output seq: null, 1, 2, 3, 3, 4, 5, 5
+ */
+private[sql] case class LastNonNull(input: Expression)
```

Comment:
`private[sql]` can be all removed per SPARK-16813

```suggestion
case class LastNonNull(input: Expression)
```